lua/kubectl/resourcebuilder.lua

```diff
@@ -196,13 +196,24 @@ function ResourceBuilder:sort()
         end
         if type(valueA) == "table" and type(valueB) == "table" then
           if valueA.timestamp and valueB.timestamp then
-            return comp(tostring(valueA.timestamp), tostring(valueB.timestamp))
+            local p = "(%d+)-(%d+)-(%d+)T(%d+):(%d+):(%d+)Z"
+            local yearA, monthA, dayA, hourA, minA, secA = valueA.timestamp:match(p)
+            local yearB, monthB, dayB, hourB, minB, secB = valueB.timestamp:match(p)
+            local epochA = os.time({ day = dayA, month = monthA, year = yearA, hour = hourA, min = minA, sec = secA })
+            local epochB = os.time({ day = dayB, month = monthB, year = yearB, hour = hourB, min = minB, sec = secB })
+            return comp(epochA, epochB)
           else
             return comp(tostring(valueA.value), tostring(valueB.value))
           end
         elseif tonumber(valueA) and tonumber(valueB) then
           return comp(valueA, valueB)
         else
+          -- sort by restarts
+          local numA = tonumber(tostring(valueA):match("(%d+)%s+%(.+ago%)"))
```

Comment:
Hmm we already have a elseif that is supposed to handle numbers, instead of bloating this sort function, I would rather handle this in in a similar way as we do the timestamp.
In the timestamp we add another field to the value ```{ value= "some value", symbol = "some hl group", timestamp = "we sort on this" }```
We could do something similar when building the content rows, when creating the Age column value we would add something like this:
```{ value= "some value", symbol = "some hl group", sort_on = "A number" }```
You would then check this in an elseif after the timestamp check.

Another option, maybe for a future rewrite, is to have a function that this method invokes instead, i.e bring you own sort

Reply:
Ok I committed https://github.com/Ramilito/kubectl.nvim/pull/205/commits/3725db3d5b625c647e35ad7b039e5130b2efd3be
can you please take a look @Ramilito  ?

Reply:
Perfect!